### PR TITLE
 Enhancements to Camera and Media Screens with Comprehensive Tests

### PR DESCRIPTION
**Pull request by Rcc999**
## **Summary**
This PR introduces comprehensive unit tests for the `CameraScreen` and `CapturedMediaScreen` components to enhance functionality verification and coverage (I manage to have a 90% coverage). Key changes include:

### **1. Added Tests**
- **`app/__tests__/CameraScreen.test.tsx`**:
  - Validates behavior for permissions, navigation, zoom functionality, and camera interactions (flash, toggle, etc.).
  - Handles edge cases like denied permissions, zoom limits, and invalid scenarios.

- **`app/__tests__/CapturedMediaScreen.test.tsx`**:
  - Covers upload functionality, header interactions, photo/video handling, and caching.
  - Tests error handling for missing/invalid dimensions, caching issues, and navigation.

image

---

### **2. File Modifications**
- **`app/screens/camera/CameraScreen.tsx`**:
  - Added `testID` props to elements for better testability.
  - Refactored button handling logic for readability and modularity.

- **`app/screens/camera/CapturedMediaScreen.tsx`**:
  - Added support for better error alerts, caching flow, and uploading media.
  - Debug alerts were introduced for critical operations like image resizing, caching, and dimension retrieval.

---

### **3. Deleted Broken Tests**
- Removed non-functional or redundant tests from unrelated components to avoid blocking CI/CD pipelines.

---

## **Testing**
- Run `npm test` to verify added functionality.
- Ensure all existing tests pass and validate the new test cases.

---

## **Notes**
- Please review the updated tests and modified files in detail to ensure compatibility with the current codebase.

 🚀

### CODE REVIEW

File: app/screens/camera/CapturedMediaScreen.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ import { ref, uploadBytes, getDownloadURL } from 'firebase/storage';  // Firebas
 import { get } from 'react-native/Libraries/TurboModule/TurboModuleRegistry';
 import { storage } from '../../../firebase/firebase'; // Import storage from your Firebase config
 import { ReportStackParamList } from '@/types/types';
```
Comment (mahdiatallah) [inline app/screens/camera/CapturedMediaScreen.tsx:9]: Are you sure that having an alias import doesn't pose any issues?

**Review by mahdiatallah — CHANGES_REQUESTED** [review]
Hi, thank you for providing tests that cover the main functionalities of the camera process. I noticed, however, that some checks are failing, particularly related to the storage tests. These seem to be outdated or erroneous tests that are already being addressed in Baudoin's PR.

Could you please remove these failing tests? Also, before merging, ensure you pull the latest changes from the main branch into your current branch. When resolving conflicts, make sure to keep the main branch's version for the screens in section 4.

Let me know if you have any questions or need clarification!

File: app/screens/camera/CapturedMediaScreen.tsx (hunk 1 above)
Comment (Rcc999) [reply app/screens/camera/CapturedMediaScreen.tsx:9]: Yes you are right, it does not cause an issue, but I will refractor it for a common purpose that we decided previously
